client/jetpack-connect/plans.jsx

```diff
@@ -210,6 +211,10 @@ class Plans extends Component {
 		// clears whatever we had stored in local cache
 		this.props.selectPlanInAdvance( null, this.props.selectedSiteSlug );
 
+		if ( cartItem.product_slug === get( this.props, 'selectedSite.plan.product_slug', null ) ) {
```

Comment:
hmm, well, it doesn't break when you select the free plan... but you are right, I've moved that `if` under the `! cartItem` check